transport/zipkin/http_test.go

```diff
@@ -125,54 +180,56 @@ func newHTTPServer(t *testing.T) *httpServer {
 		authCredentials: make([]*HTTPBasicAuthCredentials, 0),
 		mutex:           sync.RWMutex{},
 	}
-	http.HandleFunc("/api/v1/spans", func(w http.ResponseWriter, r *http.Request) {
-		contextType := r.Header.Get("Content-Type")
-		if contextType != "application/x-thrift" {
-			t.Errorf(
-				"except Content-Type should be application/x-thrift, but is %s",
-				contextType)
-			return
-		}
 
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			t.Error(err)
-			return
-		}
-		buffer := thrift.NewTMemoryBuffer()
-		if _, err = buffer.Write(body); err != nil {
-			t.Error(err)
-			return
-		}
-		transport := thrift.NewTBinaryProtocolTransport(buffer)
-		_, size, err := transport.ReadListBegin()
-		if err != nil {
-			t.Error(err)
-			return
-		}
-		var spans []*zipkincore.Span
-		for i := 0; i < size; i++ {
-			zs := &zipkincore.Span{}
-			if err = zs.Read(transport); err != nil {
+	once.Do(func() {
```

Comment:
instead of this, it's better to use [httptest.NewServer(handler)](https://golang.org/pkg/net/http/httptest/?m=all#NewServer), which avoids polluting the default mux